fdbserver/ClusterController.actor.cpp

```diff
@@ -3740,6 +3870,49 @@ void clusterRegisterMaster(ClusterControllerData* self, RegisterMasterRequest co
 	checkOutstandingRequests(self);
 }
 
+// Halts the registering (i.e. requesting) singleton if one is already in the process of being recruited
+// or, halts the existing singleton in favour of the requesting one
+template <class Interface>
+void haltRegisteringOrCurrentSingleton(ClusterControllerData* self,
+                                       const WorkerInterface& worker,
+                                       const Singleton<Interface>& currSingleton,
+                                       const Singleton<Interface>& registeringSingleton,
+                                       const Optional<UID> recruitingID) {
+	ASSERT(currSingleton.getRole() == registeringSingleton.getRole());
+	const UID registeringID = registeringSingleton.interface.get().id();
+	const std::string roleName = currSingleton.getRole().roleName;
+	const std::string roleAbbr = currSingleton.getRole().abbreviation;
+
+	// halt the requesting singleton if it isn't the one currently being recruited
+	if ((recruitingID.present() && recruitingID.get() != registeringID) ||
+	    self->clusterControllerDcId != worker.locality.dcId()) {
+		TraceEvent(("CCHaltRegistering" + roleName).c_str(), self->id)
+		    .detail(roleAbbr + "ID", registeringID)
+		    .detail("DcID", printable(self->clusterControllerDcId))
+		    .detail("ReqDcID", printable(worker.locality.dcId()))
+		    .detail("Recruiting" + roleAbbr + "ID", recruitingID.present() ? recruitingID.get() : UID());
+		if (registeringSingleton.getClusterRole() == ProcessClass::DataDistributor) {
```

Comment:
empty if block?

Reply:
Whoops this is a remnant from debugging. Removing!